lib/no_brainer/migration.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module NoBrainer
+  #
+  # Parent class of all migration scripts.
+  # This class is used by the NoBrainer::Migrator class.
+  #
+  class Migration
+    class NoImplementedError < StandardError; end
+
+    def migrate!
+      up
+    rescue StandardError
+      raise unless respond_to?(:down)
+
+      begin
+        down
```

Comment:
Print an error message right before calling down with the exception, and the fact that you are running down. People need to know what's going on.

Reply:
Okay, sure I will.